core-bundle/src/EventListener/HttpCache/StripCookiesSubscriber.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\HttpCache;
+
+use FOS\HttpCache\SymfonyCache\CacheEvent;
+use FOS\HttpCache\SymfonyCache\Events;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class StripCookiesSubscriber implements EventSubscriberInterface
+{
+    private const BLACKLIST = [
+        // Modals are always for JS only
+        '(.*)?modal(.*)?',
+
+        // Google Analytics (https://developers.google.com/analytics/devguides/collection/analyticsjs/cookie-usage)
+        '_ga',
+        '_gid',
+        '_gat',
+        'AMP_TOKEN',
+        '_gac_.+',
+
+        // Matomo (https://matomo.org/faq/general/faq_146/)
+        '_pk_id',
+        '_pk_ref',
+        '_pk_ses',
+        '_pk_cvar',
+        '_pk_hsr',
+
+        // Cloudflare
+        '__cfduid',
+    ];
+
+    /**
+     * @var array
+     */
+    private $whitelist;
+
+    public function __construct(array $whitelist = [])
+    {
+        $this->whitelist = $whitelist;
+    }
+
+    public function getWhitelist(): array
+    {
+        return $this->whitelist;
+    }
+
+    public function preHandle(CacheEvent $event): void
+    {
+        $request = $event->getRequest();
+
+        // Not a cacheable request anyway? Then we don't care
+        if (!$request->isMethodCacheable()) {
+            return;
+        }
+
+        if (!$request->cookies->count()) {
+            return;
+        }
+
+        if ($this->whitelist) {
+            $this->filterCookies($request, $this->whitelist, true);
+
+            return;
+        }
+
+        $this->filterCookies($request, self::BLACKLIST);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::PRE_HANDLE => 'preHandle',
+        ];
+    }
+
+    private function filterCookies(Request $request, array $list, bool $isWhitelist = false): void
+    {
+        $cookies = array_keys($request->cookies->all());
+
+        foreach ($cookies as $name) {
+            foreach ($list as $entry) {
+                $matches = preg_match('/^'.$entry.'$/i', $name);
```

Comment:
@ausi maybe we could create one big regex of all regexes to see if any of them matches rather than checking them one after the other. Do you think that would be faster? 🤔

Reply:
Yes, should be faster I think. Can be done with a single `preg_grep()`:

```php
$removeCookies = preg_grep(
	'/^(?:'.implode(')$|^(?:', $list).')$/i',
	array_keys($request->cookies->all()),
	$isWhitelist ? PREG_GREP_INVERT : 0
);
foreach ($removeCookies as $name) {
	$request->cookies->remove($name);
}
```

Reply:
🦊

Reply:
Jup, that was exactly what I was looking for. Just wasn't sure if this is going to be faster but I'm pretty confident it is. So implemented in da5b3904ee97d5bb6a8a8bd60a5c66f707f2e0e4.